config/300-pipeline.yaml

```diff
@@ -43,6 +41,24 @@ spec:
         # See https://kubernetes.io/blog/2019/06/20/crd-structural-schema/
         # See issue: https://github.com/knative/serving/issues/912
         x-kubernetes-preserve-unknown-fields: true
+  - name: v1
+    served: false
+    storage: false
+    schema:
+      openAPIV3Schema:
+        type: object
+        # One can use x-kubernetes-preserve-unknown-fields: true
+        # at the root of the schema (and inside any properties, additionalProperties)
+        # to get the traditional CRD behaviour that nothing is pruned, despite
+        # setting spec.preserveUnknownProperties: false.
+        #
+        # See https://kubernetes.io/blog/2019/06/20/crd-structural-schema/
+        # See issue: https://github.com/knative/serving/issues/912
+        x-kubernetes-preserve-unknown-fields: true
```

Comment:
This comment is starting to get a bit stale and I think it's sort of hard to understand. I put a more up to date explanation [here](https://github.com/tektoncd/pipeline/blob/main/config/300-task.yaml#L52-L59) but you can feel free to paraphrase or use whatever you feel is most clear.

Reply:
Thanks Lee!